packages/tools/src/tools/annotation/UltrasoundAnnotationTool/utils/segmentLargestUSOutlineFromBuffer.ts

```diff
@@ -0,0 +1,186 @@
+import type { FanShapeContour } from './types';
+
+/**
+ * Given a raw image buffer (grayscale, RGB, or RGBA) plus dimensions,
+ * returns the contour of the largest non-black region.
+ *
+ * @param {Uint8Array|Uint8ClampedArray|Array<number>} buffer
+ *   Length must be width*height (grayscale), width*height*3 (RGB), or width*height*4 (RGBA).
+ * @param {number} width   Image width in pixels
+ * @param {number} height  Image height in pixels
+ * @returns {Array<Point2>}  Clockwise contour of largest blob
+ */
+export function segmentLargestUSOutlineFromBuffer(
+  buffer,
+  width,
+  height
+): FanShapeContour {
+  const totalPixels = width * height;
+  const channelCount = buffer.length / totalPixels;
+  if (![1, 3, 4].includes(channelCount)) {
+    throw new Error('Buffer must be 1, 3, or 4 channels per pixel');
+  }
+
+  // 1) Build binary mask: true = any channel > 0
+  const mask = Array.from({ length: height }, () =>
+    new Array(width).fill(false)
+  );
+  for (let y = 0; y < height; y++) {
+    for (let x = 0; x < width; x++) {
+      const pixelIndex = y * width + x;
+      const base = pixelIndex * channelCount;
+      let isForeground = false;
+      // check R, G, B channels (or just the single channel)
+      for (let c = 0; c < Math.min(3, channelCount); c++) {
+        if (buffer[base + c] > 0) {
+          isForeground = true;
+          break;
+        }
+      }
+      mask[y][x] = isForeground;
+    }
+  }
+
+  // 2) Connected-component labeling (4-connectivity)
+  const labels = Array.from({ length: height }, () => new Array(width).fill(0));
+  let currentLabel = 0;
+  const regionSizes = {};
+
+  function floodFill(sx, sy, label) {
```

Comment:
we have already a flootFlill implementation in packages/tools/src/utilities/segmentation/floodFill.ts see if you can use that